### ISSUE TITLE
Bundle derive impl calls mem::forget on Copy types

### ISSUE THREAD

**Issue by bgourlie** [bevyengine/bevy#251]
Clippy flagged this error:

```
error: calls to `std::mem::forget` with a value that implements `Copy`. Forgetting a copy leaves the original intact.
  --> crates\bevy_render\src\entity.rs:13:10
   |
13 | #[derive(Bundle, Default)]
   |          ^^^^^^
   |
   = note: `#[deny(clippy::forget_copy)]` on by default
```

**Comment by multun** [comment]
This warning is there because some macro uses forget on a value regarless of whether it's Copy or not. This is fine, and silenced in  #206

**Comment by cart** [comment]
closed via #206
